docs/core/deploying/native-aot/fixing-warnings.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Introduction to AOT warnings
+description: Learn about why warnings may be produced when publishing application as native AOT, how to address them, and how to make the application "AOT compatible."
+author: MichalStrehovsky
+ms.author: michals
+ms.date: 09/06/2022
+---
+# Introduction to AOT warnings
+
+When publishing your application as native AOT, the build process produces all the native code and data structures required to support the application at run time. This is different from non-native deployments that execute the application from formats describing the application in abstract terms (a program for a virtual machine) and create native representations on demand at run time.
+
+The abstract representations of program parts don't have a one-to-one mapping to native representation. E.g. the abstract description of the generic `List<T>.Add` method maps to potentially infinite number of native method bodies that need to be specialized for the given T (`List<int>.Add`, `List<double>.Add`, etc.).
+
+Because the relationship of abstract code to native code is not one-to-one, the build process needs to create a complete list of native code bodies and data structures at build time. With some of the .NET APIs it can be difficult to create this list at build time. If the API is used in a way that wasn't anticipated at build time, an exception will be thrown.
```

Comment:
```suggestion
Because the relationship of abstract code to native code is not one-to-one, the build process needs to create a complete list of native code bodies and data structures at build time. It can be difficult to create this list at build time for some of the .NET APIs. If the API is used in a way that wasn't anticipated at build time, an exception will be thrown.
```